megamek/src/megamek/common/actions/WeaponAttackAction.java

```diff
@@ -789,7 +789,22 @@ else if ((te != null) && (ae.getSwarmTargetId() == te.getId())) {
 
         // if we're spotting for indirect fire, add +1 (no penalty with second pilot in command console)
         if (ae.isSpotting() && !ae.getCrew().hasActiveCommandConsole()) {
-            toHit.addModifier(+1, "attacker is spotting for indirect LRM fire");
+	    //but if it spots by TAG it doesn't.
+            boolean isSpotByTag = false;//temporal boolean flag
+            
+            //search through tagInfoForTurn and see if the entity has successfuly designated TAG on something.
+            Vector<TagInfo> tmpTagInfoForTurn = game.getTagInfo();
+            for (int i = 0; i < tmpTagInfoForTurn.size(); i++){
+                TagInfo tmpInfo = tmpTagInfoForTurn.elementAt(i);
+                if (tmpInfo.attackerId == attackerID){
```

Comment:
Java is case sensitive. The local variable you are trying to match is `attackerId`, not `attackerID`.

Reply:
Ha, what a silly mistake! Thanks for review that.